src/hal/src/window.rs

```diff
@@ -282,12 +283,31 @@ impl SwapchainConfig {
     }
 
     /// Create a swapchain configuration based on the capabilities
-    /// returned from a physical device query.
-    pub fn from_caps(caps: &SurfaceCapabilities, format: Format) -> Self {
+    /// returned from a physical device query. If the surface does not
+    /// specify a current size, default_extent is clamped and used instead.
+    pub fn from_caps(caps: &SurfaceCapabilities, format: Format, default_extent: Extent2D) -> Self {
+        let clamped_extent = match caps.current_extent {
+            Some(current) => current,
+            None => {
+                let (min_width, max_width) = (caps.extents.start.width, caps.extents.end.width - 1);
+                let (min_height, max_height) = (caps.extents.start.height, caps.extents.end.height - 1);
+
+                // clamp the default_extent to within the allowed surface sizes
+                let width = min(max_width, max(default_extent.width, min_width));
+                let height = min(max_height, max(default_extent.height, min_height));
+
+                // assert that the default_extent fits within the allowed sizes
+                debug_assert_eq!(default_extent.height, height);
```

Comment:
I don't think these assertions are needed: the parameter isn't supposed to be strict - merely a hint on what configuration to build

Reply:
I've removed the assertions